### PR TITLE
feat: show onboarding btn on small screens

### LINKED ISSUE #1462: Feat: show onboarding button on small screens
# Current behavior: 

The small screens does not show the onboarding button 

# Suggestions: 

              > I think such a design would be better than just hiding the element: 
> 
> image
> 
> 

I do like this design. Feel free to open a new issue removing the hidden flag on mobile.

_Originally posted by @bdougie in https://github.com/open-sauced/insights/issues/1460#issuecomment-1660334242_

### PR DESCRIPTION
**Pull request by 289Volts**
## Show onboarding button on small screens

This PR adds the onboarding button on mobile screens up to 640px breakpoint

## What type of PR is this? (check all applicable)

- [X] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Closes #1462

## Mobile & Desktop Screenshots/Recordings

https://github.com/open-sauced/app/assets/66239216/6a6f23e2-f62a-47e4-b120-9f44e38f5569

## Added tests?

- [ ] 👍 yes
- [X] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [X] 🙅 no documentation needed

### CODE REVIEW

**Comment by OgDev-01** [comment]
Hey @289Volts , thanks for opening a PR for this fix 👏👏... i think you have a merge conflict to resolve 👍

**Comment by 289Volts** [comment]
> Hey @289Volts , thanks for opening a PR for this fix 👏👏... i think you have a merge conflict to resolve 👍

Thank you for your feedback. I have resolved the merge conflicts and i await your honest review on the PR

**Comment by OgDev-01** [comment]
Try running the build locally to fix the build error @289Volts

**Comment by babblebey** [comment]
Hey @289Volts 

From your screencast demoing the implementation, the Onboarding button seem to not be displayed at all on Desktop/Larger screen sizes.

We wish for the button to display on these screen size too.

**Review by babblebey — COMMENTED** [review]
Hey @289Volts 

From your screencast demoing the implementation, the Onboarding button seem to not be displayed at all on Desktop/Larger screen sizes.

We wish for the button to display on these screen size too.

**Comment by 289Volts** [comment]
> Hey @289Volts
> 
> From your screencast demoing the implementation, the Onboarding button seem to not be displayed at all on Desktop/Larger screen sizes.
> 
> We wish for the button to display on these screen size too.

The reason for this is because the onboarding component is in two different sections of the header. the one that is shown on mobile is intentional to display the fix for this issue

**Comment by 289Volts** [comment]
I have also discovered that on large screens (1024px), the onboarding button text wraps to the second line because of lack of space to accommodate all the elements in the header.

localhost_3001_hub_insights

 The fix i propose is to use the same toolpicker style used for mobile screens for large screens too

localhost_3001_hub_insights (1)

OR show the tool tip when they hover on the button icon (it is accessible to screen readers)

https://github.com/open-sauced/app/assets/66239216/9ce2140d-3a25-460a-9883-031be5f5d877

**Comment by 289Volts** [comment]
> Try running the build locally to fix the build error @289Volts

I have run the build and i discovered the error wasn't caused by a component i worked on. 

Screenshot (43)

**Comment by OgDev-01** [comment]
From the above screenshot, it seems you don't have the `@nivo/bar` package installed. crosscheck to confirm you have 
image
in the `package.json` file. if no, then try installing it with `npm i @nivo/bar` else, just run `npm ci` to update your dependencies

**Review by OgDev-01 — APPROVED** [review]
This workd 🍕

File: .eslintrc.json [hunk 1]
```diff
@@ -2,6 +2,9 @@
   "extends": ["next/core-web-vitals", "plugin:prettier/recommended", "plugin:storybook/recommended"],
   "plugins": ["unused-imports"],
   "rules": {
+    "prettier/prettier": ["error",{
+      "endOfLine": "auto"}
+    ],
```
Comment (brandonroberts) [inline .eslintrc.json:7]:
```suggestion
    "prettier/prettier": ["error",{"endOfLine": "auto"}],
```

File: components/molecules/AuthSection/auth-section.tsx [hunk 2]
```diff
@@ -36,6 +36,7 @@ const AuthSection: React.FC = ({}) => {
   const [userInfo, setUserInfo] = useState<DbUser | undefined>(undefined);
   const [host, setHost] = useState<string>("");
 
+  console.log(user, onboarded);
```
Comment (brandonroberts) [inline components/molecules/AuthSection/auth-section.tsx:39]:
```suggestion
```

File: components/molecules/AuthSection/auth-section.tsx [hunk 3]
```diff
@@ -104,9 +105,35 @@ const AuthSection: React.FC = ({}) => {
             <>
               {!onboarded && (
                 <>
-                  <OnboardingButton />
+                  <div className="relative hidden lg:flex xl:hidden">
+                    <Tooltip content="Complete the onboarding" direction="bottom">
+                      <OnboardingButton ariaLabel="Click to complete your onboarding" className="!flex !pr-1" />
+                    </Tooltip>
+                    {/* <OnboardingButton aria="onboarding" className="!flex !pr-1">
```
Comment (brandonroberts) [inline components/molecules/AuthSection/auth-section.tsx:112]: Delete unused code

File: components/molecules/AuthSection/auth-section.tsx [hunk 4]
```diff
@@ -104,9 +105,35 @@ const AuthSection: React.FC = ({}) => {
...
+                    </div> */}
+                  </div>
+                  <OnboardingButton className="!hidden xl:!flex !pr-1">
+                    <Text className="text-sm !text-light-slate-12 hidden xl:block py-1 px-2">
+                      Complete the onboarding
+                    </Text>
+                  </OnboardingButton>
                   <Divider type="vertical" className="!h-6 !bg-light-slate-6"></Divider>
                 </>
+                // <>
```
Comment (brandonroberts) [inline components/molecules/AuthSection/auth-section.tsx:131]: Delete unused code

File: components/molecules/OnboardingButton/onboarding-button.tsx [hunk 5]
```diff
@@ -1,15 +1,25 @@
...
+      aria-labelledby={aria}
+      aria-label={ariaLabel}
+      onClick={() => router.push("/start")}
+      className={`${className} hidden items-center gap-2 border border-light-orange-8 py-1 px-1 pr-3 rounded-lg md:flex`}
+    >
       <ProgressPie percentage={66} />
-      <Text className="text-sm !text-light-slate-12  ">
-        <Link href="/start">Complete the onboarding</Link>
-      </Text>
+      {children && children}
```
Comment (brandonroberts) [inline components/molecules/OnboardingButton/onboarding-button.tsx:22]: Why not just `children`?

**Review by brandonroberts — CHANGES_REQUESTED** [review]

**Review by brandonroberts — APPROVED** [review]

**Pull request merged**